Deprecate: Mark AppBarTheme & AppBarThemeData color parameter as deprecated in favor of backgroundColor

rkishan516 opened the pull request:
Deprecate: Mark AppBarTheme color parameter as deprecated in favor of backgroundColor
fixes: #169918

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

rkishan516 committed:
Deprecate: Mark AppBarTheme color parameter as deprecated in favor of backgroundColor

QuncCccccc reviewed (commented):
Thank you for your contribution! Since AppbarThemeData also has this property now, maybe we can deprecate it as well:)

We might also want to add tests in `/Users/quncheng/Desktop/flutter/packages/flutter/test_fixes/material/app_bar_theme.dart` and `/Users/quncheng/Desktop/flutter/packages/flutter/test_fixes/material/app_bar_theme.dart.expect` to make sure the dart fix works.

rkishan516 committed:
Deprecate: Mark AppBarThemeData color parameter as deprecated in favor of backgroundColor

rkishan516 committed:
Test: Add data driven test for color in app bar theme

rkishan516 committed:
Test: Add data driven test for color in app bar theme data

rkishan516 commented:
> Thank you for your contribution! Since AppbarThemeData also has this property now, maybe we can deprecate it as well:)
> 
> We might also want to add tests in `/Users/quncheng/Desktop/flutter/packages/flutter/test_fixes/material/app_bar_theme.dart` and `/Users/quncheng/Desktop/flutter/packages/flutter/test_fixes/material/app_bar_theme.dart.expect` to make sure the dart fix works.

Thanks for review @QuncCccccc, I have added deprecation for AppBarThemeData and also added tests as per your request.

chunhtai reviewed (approved):
LGTM, except we need a migration guide in flutter/website

QuncCccccc commented:
> we need a migration guide in flutter/website

@chunhtai Just have a question, if we are not actively removing the deprecated APIs, just marking them as deprecated, do we still need the migration guide?

QuncCccccc reviewed (approved):
LGTM. Sorry I thought I had approved it:)

chunhtai commented:
> @chunhtai Just have a question, if we are not actively removing the deprecated APIs, just marking them as deprecated, do we still need the migration guide?

yes, marking deprecation should be accompanied with a migration guide.

rkishan516 commented:
@chunhtai I have created migration guide in [website PR](https://github.com/flutter/website/pull/12156).

QuncCccccc committed:
Merge branch 'master' into app-bar-theme-color
